scripts/validate_examples/validate_examples.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+"""Validate example_pipelines modules by compiling every exported pipeline."""
+
+from __future__ import annotations
+
+import argparse
+import importlib.util
+import sys
+import tempfile
+import traceback
+import warnings
+from pathlib import Path
+from types import ModuleType
+from typing import List, Sequence, Tuple
+
+from kfp import compiler, dsl
+
+from ..utils import find_pipeline_functions, get_repo_root, normalize_targets
+
+REPO_ROOT = get_repo_root()
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description=(
+            "Import example_pipelines.py modules for the specified components or pipelines "
+            "and compile every @dsl.pipeline function they export."
+        )
+    )
+    parser.add_argument(
+        "paths",
+        metavar="PATH",
+        nargs="*",
+        help=(
+            "Component or pipeline directories (or files within them). "
+            "If omitted, every example_pipelines.py file is validated."
+        ),
+    )
+    return parser.parse_args()
+
+
+def discover_example_files(targets: Sequence[Path]) -> List[Path]:
+    discovered: List[Path] = []
+
+    for target in targets:
+        search_root = target if target.is_dir() else target.parent
+
+        for candidate in search_root.rglob("example_pipelines.py"):
+            if candidate in discovered or not candidate.is_file():
+                continue
+            try:
+                relative = candidate.relative_to(REPO_ROOT)
+            except ValueError:
+                warnings.warn(
+                    f"Unable to determine relative path for {candidate} "
+                    f"relative to repo root {REPO_ROOT}. Skipping.",
+                )
+                continue
+            if relative.parts and relative.parts[0] in {"components", "pipelines"}:
+                discovered.append(candidate)
+
+    return discovered
+
+
+def load_module_from_path(module_path: Path) -> ModuleType:
+    relative = module_path.relative_to(REPO_ROOT)
+    sanitized = "_".join(relative.with_suffix("").parts)
+    module_name = f"example_pipelines__{sanitized}"
+
+    spec = importlib.util.spec_from_file_location(module_name, module_path)
+    if spec is None or spec.loader is None:  
+        raise ImportError(f"Unable to load module from {module_path}")
+
+    module = importlib.util.module_from_spec(spec)
+    sys.modules[module_name] = module
+    spec.loader.exec_module(module)
```

Comment:
I'm not sure a way around it given the problem statement this addresses, but this (as well as the compile check) makes me a bit uneasy - a bad actor could include arbitrary malicious code in a `example_pipeline.py`'s `dsl.pipeline` function of a new component.  They could then open a PR, and this block would directly import (and therefore execute) it on our github runners.  The failsafe is pretty fragile - I believe we're relying on the fact that CI is only run automatically (without /ok-to-test) for vetted community members to assume these contributions are 'safe' to run CI on

Reply:
we discussed offline that there is sufficient guards, ownership and permissions in place surrounding this github workflows to ensure no sensitive data could be leaked in this workflow via the vectors mentioned above. no-op.